test/integration/etl_test.py

```diff
@@ -33,7 +33,7 @@ def test_pudl_engine(
 
     if check_foreign_keys:
         # Raises ForeignKeyErrors if there are any
-        pudl_sql_io_manager.check_foreign_keys()
+        pudl_sql_io_manager._sqlite_io_manager.check_foreign_keys()
```

Comment:
This is dirty, but I'm not sure what would be the best way around it. Maybe implement `MixedFormatIOManager.get_sqlite_io_manager() -> PudlSqliteIOManager` method that we could call when we really need to access sqlite only features?

Reply:
I think implementing a getter that throws a helpful error if the `MixedFormatIOManager` doesn't have a `sqlite_io_manager`.

It might not be worth the complexity but we could make the `MixedFormatIOManager` more generalizable and accept an arbitrary number of IO managers to use. Then we could have some methods for viewing and grabbing specific io managers:

```python
>>> MixedFormatIOManager.list_io_manager_names()
["sqlite_io_manager", "parquet_io_manager", "bq_io_manager", ...]
>>> MixedFormatIOManager.get_io_manager("sqlite_io_manager")
```

I think for now a simple `MixedFormatIOManager.get_sqlite_io_manager() -> PudlSqliteIOManager` method seems fine.

Reply:
I think the generalization here is probably not warranted. I would only add that if/when we know we will need it but otherwise it may be just future proofing for something that will never happen and I agree that the complexity is not worth it.

What this does show, however, is that sqlite and pudl-sqlite io managers are special in that their APIs and functionality is much broader than that of the typical/format-agnostic io manager and this API is a bit vague.

It might be valuable to separate these concerns somewhat, perhaps by defining the interface that such sqlite thingy should provide and have a method on the mixed io manager that can return implementations of this clearly-defined and separate interface. It could return the io_manager that implements this interface, but in most cases I see, we don't actually need the io_manager but the specific sqlite validation/schema-management functionality.

It might even make sense to separate this sqlite/schema functionality to a standalone class that is embedded in the IO manager but with which IO manager communicated through this well defined API.

Note that the above may very well go past the scope of this PR and should be likely reserved to subsequent cleanup/refactoring. Providing `get_sqlite_io_manager()` function is simple and deals with this dirtiness reasonably well.